routers/repo/repo.go

```diff
@@ -151,8 +151,10 @@ func Create(ctx *context.Context) {
 		}
 	}
 
-	if !ctx.User.CanCreateRepo() {
+	if ctx.Data["private"] == false && !ctx.User.CanCreateRepo() {
 		ctx.RenderWithErr(ctx.Tr("repo.form.reach_limit_of_creation", ctx.User.MaxCreationLimit()), tplCreate, nil)
+	} else if ctx.Data["private"] == true && !ctx.User.CanCreatePrivateRepo() {
+		ctx.RenderWithErr(ctx.Tr("repo.form.reach_limit_of_private_creation", ctx.User.MaxPrivateCreationLimit()), tplCreate, nil)
```

Comment:
Ah so you intend to split the repo creation limits so that MaxCreationLimit only applies to public repositories and MaxPrivateCreationLimit only applies to private repositories. 

This will essentially double users repo creation limits. You would be better off having a third limit MaxPublicCreationLimit.

You also need to handle what happens when a public repo becomes private and vice versa - see `models/repo.go:1417:func updateRepository(...)`.

There is also the issue of ForkedRepos when their base repos have their visibility changed to private then the fork has to become private. How does that affect this?

Reply:
Thanks for the comments, I'm still working through the source code to see what's arranged where... and it was 02:30 when I was committing this. :rofl: 

My initial thought was to split them. You want to have a limit for private, public and total? That would mean a user could have 2 private, 2 public and 3 total for example... leading to faulty settings and yet another error case needing to be checked. Or is that not what you meant?

As for the ForkedRepos.... When a user forks a repo, doesn't the fork become a mostly separate copy in its own right? I haven't tested it, but I'd be surprised if, on GitHub, a forked repo suddenly becomes private if the original repo does so. I don't really see the point of forcing forks to become private since, instead of forking, someone could just clone the original when it was still public and then push it as a new public repo with the same content as the original. Effectively bypassing the "forked" status.

Reply:
>  You want to have a limit for private, public and total? That would mean a user could have 2 private, 2 public and 3 total for example... leading to faulty settings and yet another error case needing to be checked.

I would read that is: A user would be allowed 3 repositories in total: 2 private, 1 public or 1 private, 2 public - up to the user how that's split - it would mean that you could make things private or public as necessary.

> As for the ForkedRepos.... 

On Gitea if the parent repo becomes private - your fork will become private. You can break the fork relationship, and then make it public if you like. 

This is particularly important if you have forked an organizations repos - a fork of a private organisation's repos is private even if the repo is a "public" repo of that organization. So if you make an organization private then all repos forked from its repos will become private.